### PR TITLE
Fix crash on scroll position restoration

### LINKED ISSUE #1604: Crash in emoji-search after scroll position preservation commit
I haven't properly reviewed the PR yet, but I'm noticing a crash in Emoji Search after these changes:

- Run Emoji Search using Android Views
- Type "medal" into the search box
- Click on one of the medal emojis
- Press the system back button
- The app crashes with the exception `app.cash.zipline.ZiplineException: IllegalArgumentException: Failed requirement.`

_Originally posted by @veyndan in https://github.com/cashapp/redwood/issues/1599#issuecomment-1766197162_

### PR DESCRIPTION
**Pull request by jingwei99**
Fixes: https://github.com/cashapp/redwood/issues/1604

### CODE REVIEW

File: redwood-lazylayout-view/src/main/kotlin/app/cash/redwood/lazylayout/view/ViewLazyList.kt [hunk 1]
```diff
@@ -107,7 +107,7 @@ internal open class ViewLazyList private constructor(
             val viewsOnScreen = recyclerView.children.map { recyclerView.getChildViewHolder(it).bindingAdapterPosition }
             userHasScrolled = true // Prevent guest code from hijacking the scrollbar.
             onViewportChanged?.invoke(
-              viewsOnScreen.min(),
+              viewsOnScreen.min().coerceAtLeast(0),
```
Comment (jingwei99) [inline redwood-lazylayout-view/src/main/kotlin/app/cash/redwood/lazylayout/view/ViewLazyList.kt:110]: `.min()` _sometimes_ returns -1 (which had been observed / is expected)

**Comment by jingwei99** [comment]
I need another round of internal release when this is in.

**Review by JakeWharton — APPROVED** [review]

**Review by swankjesse — COMMENTED** [review]
LGTM

File: redwood-lazylayout-view/src/main/kotlin/app/cash/redwood/lazylayout/view/ViewLazyList.kt (hunk 1 above)
Comment (JakeWharton) [reply redwood-lazylayout-view/src/main/kotlin/app/cash/redwood/lazylayout/view/ViewLazyList.kt:110]: -1 is `NO_POSITION` which is what is returned for all positions between adapter invalidations and the next layout pass.

Also `min` and `max` throw if the sequence is empty, are we assuming the scroll callback will never be invoked on an empty list?

Comment (jingwei99) [reply redwood-lazylayout-view/src/main/kotlin/app/cash/redwood/lazylayout/view/ViewLazyList.kt:110]: > Also min and max throw if the sequence is empty, are we assuming the scroll callback will never be invoked on an empty list?

ahhh 🤔, that's the case (not invoked on empty list) so far (this has only only been applied to emoji search). But it's not guaranteed the invocation wouldn't happen on an empty list (someone else might use it, or other teams might use it).  

> -1 is NO_POSITION which is what is returned for all positions between adapter invalidations and the next layout pass.

👍

**Pull request merged**